### ISSUE TITLE
os.IsNotExists doesn't recognize ENOENT path error from ioutil→os.ReadFile

### ISSUE THREAD

**Issue by dmke** [golang/go#44349]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

As of writing, Go 1.16 is the latest release.

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/dm/.cache/go-build"
GOENV="/home/dm/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/dm/code/go/pkg/mod"
GONOPROXY="[omitted]"
GONOSUMDB="[omitted]"
GOOS="linux"
GOPATH="/home/dm/code/go"
GOPRIVATE="[omitted]"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/dm/.local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/dm/.local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/dm/code/go-uci/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build122615617=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

While reducing some linter warnings in [go-uci](https://github.com/digineo/go-uci), I've stumbled upon a small difference going from Go 1.15 to 1.16.

In a [loadConfig](https://github.com/digineo/go-uci/blob/cd86efb9cb51aae3169300b8a8f888542a04a96e/uci.go#L108) method, I'm using `ioutil.ReadFile`, and in a [corresponding test](https://github.com/digineo/go-uci/blob/cd86efb9cb51aae3169300b8a8f888542a04a96e/uci_test.go#L72), I'm checking for ENOENT via `os.IsNotExists`. This test stopped working with Go 1.16.

Basically:

```go
_, err := ioutil.ReadFile("nonexistent")
os.IsNotExist(err) //=> true
errors.Is(err, os.ErrNotExist) //=> true
```

In Go 1.16, `os.IsNotExist` reports false.

### What did you expect to see?

See above.

### What did you see instead?

See above.

**Comment by dmke** [comment]
Hm.

While trying to create a self-contained example, I've discovered the following:

```go
package main

import (
	"errors"
	"fmt"
	"io/ioutil"
	"os"
)

func main() {
	_, err := ioutil.ReadFile("/nonexistent")
	fmt.Println("os.IsNotExist", os.IsNotExist(err))
	fmt.Println("errors.Is", errors.Is(err, os.ErrNotExist))

	for ; err != nil; err = errors.Unwrap(err) {
		fmt.Printf("%#v (%T)\n", err, err)
	}
}
```

```console
$ GOROOT=~/go-1.16 ~/go-1.16/bin/go run test.go
os.IsNotExist true
errors.Is true
&fs.PathError{Op:"open", Path:"/nonexistent", Err:0x2} (*fs.PathError)
0x2 (syscall.Errno)

$ GOROOT=~/go-1.15.8 ~/go-1.15.8/bin/go run test.go
os.IsNotExist true
errors.Is true
&os.PathError{Op:"open", Path:"/nonexistent", Err:0x2} (*os.PathError)
0x2 (syscall.Errno)
```

Not sure what I've been seeing earlier... Sorry for the noise.
